src/EditorFeatures/Core/GoToDefinition/GoToDefinitionHelpers.cs

```diff
@@ -101,6 +101,13 @@ internal static class GoToDefinitionHelpers
             bool thirdPartyNavigationAllowed = true,
             bool throwOnHiddenDefinition = false)
         {
+            var languageServices = project.Solution.Workspace.Services.GetLanguageServices(LanguageNames.FSharp);
```

Comment:
1. All this code should likely be gated by `thirdPartyNavigationAllowed` right?
2. You shouldn't look for FSharp specifically.  Instead, we should be importing this type as just a MEF service.  This would help us find *all* possible 3rd parties that want to take over navigation.

Reply:
Regarding (2), which method should I be calling to achieve this?

Reply:
I think I got it. I'll push changes soon.